wazo_dird/database/queries/phonebook.py

```diff
@@ -303,35 +335,61 @@ def _get_contact(
 
     def _get_phonebook(
         self,
-        s: Session,
+        s: BaseSession,
         visible_tenants: builtins.list[str] | None,
         phonebook_key: PhonebookKey,
     ) -> Phonebook:
-        phonebook_filter = phonebook_key_to_predicate(phonebook_key)
-        if visible_tenants is None:
-            return phonebook_filter
-        elif not visible_tenants:
-            return text('false')
-        else:
-            filter_ = and_(phonebook_filter, Phonebook.tenant_uuid.in_(visible_tenants))
-            phonebook = s.query(Phonebook).filter(filter_).first()
-            if not phonebook:
-                raise NoSuchPhonebook(cast(dict, phonebook_key))
-            return phonebook
+        filter_ = phonebook_selector(visible_tenants, phonebook_key)
+
+        phonebook = s.query(Phonebook).filter(filter_).first()
+        if not phonebook:
+            raise NoSuchPhonebook(
+                cast(dict, phonebook_key), tenants_in_scope=visible_tenants
+            )
+        return phonebook
 
-    def _list_contacts(
+    def _query_contacts(
         self,
-        s: Session,
-        query: ColumnElement,
+        s: BaseSession,
         phonebook_key: PhonebookKey,
-        search: str | None,
-    ):
+        search: str | None = None,
+    ) -> Query:
         phonebook_filter = contact_phonebook_key_to_predicate(phonebook_key)
```

Comment:
A bit out of scope, but I feel we have two words for the same thing: `filter` and `predicate` refer to the same notion, right? Can we use only one word?